src/python/pants/bin/local_pants_runner.py

```diff
@@ -220,11 +221,22 @@ def set_start_time(self, start_time):
     self._run_start_time = start_time
     self._reporting.initialize(self._run_tracker, self._options, start_time=self._run_start_time)
 
+    target_specs = [self.normalise_spec(spec) for spec in self._options.target_specs]
+    self._run_tracker.run_info.add_info("specs", target_specs, stringify=False)
+
     # Capture a repro of the 'before' state for this build, if needed.
     self._repro = Reproducer.global_instance().create_repro()
     if self._repro:
       self._repro.capture(self._run_tracker.run_info.get_as_dict())
 
+  @staticmethod
+  def normalise_spec(spec):
+    if spec.endswith("::"):
+      if spec.startswith("//") and not spec.startswith("//:"):
```

Comment:
Rather than this parsing, you should probably use https://github.com/pantsbuild/pants/blob/d4676d1829b13c924ff7b2e775dc0ab90f469d21/src/python/pants/base/cmd_line_spec_parser.py#L51-L57 followed by https://github.com/pantsbuild/pants/blob/d4676d1829b13c924ff7b2e775dc0ab90f469d21/src/python/pants/base/specs.py#L25-L27